debug/process/debugger.go

```diff
@@ -158,10 +160,31 @@ func dumpGoRoutines() {
 }
 
 func (debugger *processDebugger) changeLog() {
+	oldLogLevel := logger.GetLogLevelPattern()
+
 	errSetLogLevel := logger.SetLogLevel(debugger.debuggingLogLevel)
 	if errSetLogLevel != nil {
-		log.Error("debugger.changeLog: cannot change log level", "error", errSetLogLevel)
+		log.Error("processDebugger: cannot change log level", "error", errSetLogLevel)
+	}
+
+	if debugger.revertTimeInSeconds > 0 {
+		debugger.cancel()
+
+		go debugger.revertLogLevel(oldLogLevel)
+	}
+}
+
+func (debugger *processDebugger) revertLogLevel(oldLogLevel string) {
+	timeToWait := time.Second * time.Duration(debugger.revertTimeInSeconds)
+	log.Debug("processDebugger revertLogLevel", "original log level", oldLogLevel, "will revert in", timeToWait)
+
+	time.Sleep(timeToWait)
+
+	errSetLogLevel := logger.SetLogLevel(oldLogLevel)
+	if errSetLogLevel != nil {
+		log.Error("processDebugger changeLog: cannot change log level", "error", errSetLogLevel)
```

Comment:
changeLog or revertLogLevel? changeLog is the name of the above method

Reply:
right, too much copy paste